SMGFork: Added simple visualisation of the analysis of CPAlien

viktormalik opened the pull request:
The visualisation is done via HTML files.

viktormalik committed:
SMGFork: Added simple visualisation of the analysis of CPAlien via HTML files.

viktormalik committed:
SMGFork: Highlighting of added elements from previous SMG in visualisation.

petr-muller commented on scripts/smg-html.py:
```diff
@@ -0,0 +1,208 @@
+# create directories if don't exist
+dirMain = "../output/smg-html"
+dirImg = "../output/smg-html/img"
+dirOutput = "../output"
```
The base path (../output) should be configurable (passed as a parameter) rather than hardcoded. The script should not depend on call location.

petr-muller commented on scripts/smg-html.py:
```diff
@@ -0,0 +1,208 @@
+    filePath = os.path.join(dirMain, fileName)
+    try:
+        if os.path.isfile(filePath):
+            os.unlink(filePath)
```
The whole thing could be simplified by using shutil.rmtree and then re-creating dirs again

petr-muller commented on scripts/smg-html.py:
```diff
@@ -0,0 +1,208 @@
+        if os.path.isfile(filePath):
+            os.unlink(filePath)
+    except Exception, e:
+        raise e
```
Why catch and reraise?

petr-muller commented on scripts/smg-html.py:
```diff
@@ -0,0 +1,208 @@
+        if os.path.isfile(filePath):
+            os.unlink(filePath)
+    except Exception, e:
+        raise e
```
Again, just remove it all and recreate. Saves code.

petr-muller commented on scripts/smg-html.py:
```diff
@@ -0,0 +1,208 @@
+        filePath = os.path.join(dirOutput, fileName)
+        (name, ext) = os.path.splitext(fileName)
+        destPath = os.path.join(dirImg, name + ".png")
+        dotCmd = "dot -Tpng " + filePath + " -o " + destPath
```
The graphs are better to present as vector using preferably SVG.

petr-muller commented on scripts/smg-html.py:
```diff
@@ -0,0 +1,208 @@
+        (name, ext) = os.path.splitext(fileName)
+        destPath = os.path.join(dirImg, name + ".png")
+        dotCmd = "dot -Tpng " + filePath + " -o " + destPath
+        os.system(dotCmd)
```
subprocess module is always preferred to os.system

petr-muller commented on scripts/smg-html.py:
```diff
@@ -0,0 +1,208 @@
+#     if smg.succ:
+#         for succ in smg.succ:
+#             print succ
+#     print "\n"
```
Please do not commit commented-out code. Such code can either be part of some debugging output (triggered by some debug option) or not checked in at all.

petr-muller commented on src/org/sosy_lab/cpachecker/cpa/smgfork/SMGTransferRelation.java:
```diff
@@ -147,6 +152,11 @@
+   * Path to the file where analysis tree for visualisation is stored
+   */
+//  private Path analysisSchema;
+
```
Please do not commit commented-out code.

petr-muller commented on src/org/sosy_lab/cpachecker/cpa/smgfork/SMGTransferRelation.java:
```diff
@@ -509,15 +553,15 @@ public SMGTransferRelation(Configuration config, LogManager pLogger,
       mallocFailState = null;
-    } else {
-      successor.setPredecessor(smgState);
+    } else {successor.setPredecessor(smgState);
```
Incorrect indent

petr-muller commented on src/org/sosy_lab/cpachecker/cpa/smgfork/SMGTransferRelation.java:
```diff
@@ -509,15 +553,15 @@ public SMGTransferRelation(Configuration config, LogManager pLogger,
     }
-
-    for (SMGState smg : result) {
+ for (SMGState smg : result) {
```
Incorrect indent.

viktormalik committed:
SMGFork: Repaired errors in HTML visualisation script

viktormalik committed:
SMGFork: Repaired bold fonts in visualisation of recently added items.

viktormalik committed:
SMGFork: Remade visualisation of stack frames in dot from record nodes to HTML-style nodes.
Added basic support for tracking removed SMG elements in SMG and CLangSMG classes. Tracking is not done in SMGPlotter yet.

viktormalik committed:
SMGFork: Added visualisation of removed elements between two SMGs.

petr-muller commented:
:wave:
